enhancements/olm/olm-managed-operator-metrics.md

```diff
@@ -0,0 +1,145 @@
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+
+## Summary
+
+The primary purpose of this enhancement is to propose a set of changes that would enable [Operator-Lifecycle-Manager (OLM)](https://github.com/operator-framework/operator-lifecycle-manager) managed operators to define `ServiceMonitors` and `PrometheusRule` objects to interact with the [Prometheus Operator](https://github.com/coreos/prometheus-operator). This feature could then be used on OpenShift clusters to expose metrics to [Telemeter](https://github.com/openshift/telemeter).
+
+For the uninitiated, Red Hat uses metrics collected by Telemeter to:
+
+- Proactively reach out to customers with "problem clusters"
+- Troubleshoot customer issues
+- Understand how individual applications are behaving at a wholistic level
+
+## Motivation
+
+OLM managed operators would significantly benefit from the ability to interact with Prometheus Operators deployed on cluster. By presenting OLM managed operators with the ability to seamlessly integrate with the Prometheus Operator, Cluster Admins would be given a single portal to gauge the health of operators installed on their clusters.
+
+This feature present unique value on OpenShift Clusters, where Telemeter could provide Red Hat with the ability to identify and proactively fix customer issues caused by Red Hat owned operators.
+
+### Goals
+
+- Provide operators with a first-class way of creating `ServiceMonitors` objects
+- Provide operators with a first-class way of creating `PrometheusRules` objects
+- Provide operators with a first-class way of specifying a namespace that they must be deployed to
+
+### Non-Goals
+
+- Provide generic metrics about OLM managed operators
+
+## Proposal
+
+### OLM support for new object in bundles
+
+The bundle object currently has the following format:
+
+```bash
+$ tree
+bundle
+    ├── manifests
+    │   ├── example.crd.yaml
+    │   ├── example.csv.yaml
+    │   └── example.rbac.yaml
+    └── metadata
+        └── annotations.yaml
+```
+
+Notice that OLM understands how to create and manage CSVs, CRDs, and RBAC files. This enhancement proposes expanding this feature to support `Namespaces`, `ServiceMonitors` and `PrometheusRules` objects:
+
+```bash
+$ tree
+bundle
+    ├── manifests
+    │   ├── example.crd.yaml
+    │   ├── example.csv.yaml
+    │   ├── example.ns.yaml
+    │   ├── example.prometheusrule.yaml
+    │   ├── example.rbac.yaml
+    │   └── example.servicemonitor.yaml
+    └── metadata
+        └── annotations.yaml
+```
+
+With this change, OLM will need to be updated to manage the `Namespace`, `ServiceMonitor` and `PrometheusRule` objects through the lifecycle of an operator.
+
+#### Namespace Support
+
+As OLM gains support for `Namespace` objects, The `ClusterServiceVersion` will be updated to include an optional `NamespaceRequirements` object field. The `NamespaceRequirements` will look like the following:
```

Comment:
What happens if the cluster-admin decides to NOT install the operator in that namespace. I think this is a scenario that we have to account for.

Reply:
I agree - is this a feature that we need for the first release of this feature. Solution could involve creating a subscription override as suggested [here](https://github.com/openshift/enhancements/pull/152#discussion_r357657343).

Reply:
if we're calling it "NamespaceRequirements" then you shouldn't be able to install it elsewhere.  If we're going to let you install it elsewhere, the name of the struct needs to change.